### PR TITLE
Add missing is_ad parameters

### LINKED ISSUE #15571: Properly declare derivatives as dual numbers in ADDerivativeParsedMaterial
## Reason
I missed a `TODO:` in #15466

## Design
Add missing `declareGenericPropertyDerivative`

## Impact
Better derivative information. Tag @tophmatthews

### PR DESCRIPTION
**Pull request by dschwen**
Ok, things weren't as bad as I initially thought. It's just a small change.

Closes #15571

### CODE REVIEW

**Comment by dschwen** [comment]
The "have material property" tests for block and boundary restricted materials were not AD'ified. Added a fix to make that happen.

File: test/include/kernels/DefaultMatPropConsumerKernel.h [hunk 1]
```diff
@@ -24,3 +24,26 @@ class DefaultMatPropConsumerKernel : public DerivativeMaterialInterface<Kernel>
...
+
+protected:
+  virtual GenericReal<is_ad> computeQpResidual() { return 0.0; };
+
+  const GenericMaterialProperty<Real, is_ad> & _prop;
+};
+
+typedef DefaultMatPropConsumerKernelTempl<false> DefaultMatPropConsumerKernel;
+typedef DefaultMatPropConsumerKernelTempl<true> ADDefaultMatPropConsumerKernel;
+ */
```
Comment (loganharbour) [inline test/include/kernels/DefaultMatPropConsumerKernel.h:49]: Whatcha doin?

File: test/src/kernels/DefaultMatPropConsumerKernel.C [hunk 2]
```diff
@@ -24,3 +24,26 @@ DefaultMatPropConsumerKernel::DefaultMatPropConsumerKernel(const InputParameters
...
+}
+
+template <bool is_ad>
+DefaultMatPropConsumerKernelTempl<is_ad>::DefaultMatPropConsumerKernelTempl(
+    const InputParameters & parameters)
+  : DerivativeMaterialInterface<GenericKernel<is_ad>>(parameters),
+    _prop(getDefaultMaterialProperty<Real>("mat_prop"))
+{
+}
+ */
```
Comment (loganharbour) [inline test/src/kernels/DefaultMatPropConsumerKernel.C:49]: Really though - what's up?

**Review by loganharbour — CHANGES_REQUESTED** [review]

File: test/include/kernels/DefaultMatPropConsumerKernel.h (hunk 1 above)
Comment (dschwen) [reply test/include/kernels/DefaultMatPropConsumerKernel.h:49]: Could you phrase your question a little more specific?

Comment (dschwen) [reply test/include/kernels/DefaultMatPropConsumerKernel.h:49]: Oh, I see a leftover commented piece of code. I was waiting for the GenericKernel PR. If that's in I can simplify the code.

Comment (loganharbour) [reply test/include/kernels/DefaultMatPropConsumerKernel.h:49]: Got it. Do you want this in first?

**Comment by dschwen** [comment]
It is in. I'm testing it now.

**Review by loganharbour — APPROVED** [review]

File: test/tests/materials/derivative_material_interface/ad_material_chaining.i [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+  [../]
+  [./D2]
+    type = ADElementIntegralMaterialProperty
+    mat_prop = D2
+  [../]
+  [./E0]
+    type = ADElementIntegralMaterialProperty
+    mat_prop = E0
+  [../]
+[]
```
Comment (lindsayad) [inline test/tests/materials/derivative_material_interface/ad_material_chaining.i:140]: @dschwen the gold file for these just has all zeros. Is that really what it should be?

**Pull request merged**